apps/hash-frontend/src/pages/inbox.page.tsx

```diff
@@ -0,0 +1,225 @@
+  borderWidth: 1,
+  overflow: "hidden",
+}));
+
+const TableRow = styled(MuiTableRow)(() => ({
+  [`&:last-child .${tableCellClasses.body}`]: {
+    borderBottom: "none",
+  },
+  [`&:first-child .${tableCellClasses.head}`]: {
+    "&:first-child": {
+      borderTopLeftRadius: "8px",
+    },
+    "&:last-child": {
+      borderTopRightRadius: "8px",
+    },
+  },
+  [`&:last-child .${tableCellClasses.body}`]: {
+    "&:first-child": {
+      borderBottomLeftRadius: "8px",
+    },
+    "&:last-child": {
+      borderBottomRightRadius: "8px",
+    },
+  },
+}));
+
+const TableCell = styled(MuiTableCell)(({ theme }) => ({
+  whiteSpace: "nowrap",
+  border: 0,
+  borderStyle: "solid",
+  borderColor: theme.palette.gray[20],
+  borderBottomWidth: 1,
+  borderRightWidth: 1,
+  padding: theme.spacing(0.5, 1.5),
+  [`&.${tableCellClasses.head}`]: {
+    fontSize: 13,
+    fontWeight: 600,
+    color: theme.palette.common.black,
+  },
+  [`&.${tableCellClasses.body}`]: {
+    fontSize: 14,
+    fontWeight: 500,
+    color: theme.palette.gray[90],
+  },
+  "&:last-child": {
+    borderRightWidth: 0,
+  },
+}));
+
+const NotificationRow: FunctionComponent<Notification> = (notification) => {
+  const { markNotificationAsRead } = useNotifications();
+  const { kind, triggeredByUser, occurredInPage, occurredInBlock } =
+    notification;
+
+  const ownedById = useMemo(
+    () =>
+      extractOwnedByIdFromEntityId(occurredInPage.metadata.recordId.entityId),
+    [occurredInPage],
+  );
+
+  const { shortname } = useUserOrOrgShortnameByOwnedById({ ownedById });
+
+  const pageHref = useMemo(() => {
+    if (!shortname) {
+      return undefined;
+    }
+
+    /** @todo: append query param if the mention was in a comment */
+    return constructPageRelativeUrl({
+      workspaceShortname: shortname,
+      pageEntityUuid: extractEntityUuidFromEntityId(
+        occurredInPage.metadata.recordId.entityId,
+      ),
+      highlightedBlockEntityId: occurredInBlock.metadata.recordId.entityId,
+    });
+  }, [shortname, occurredInPage, occurredInBlock]);
+
+  const pageTitle = useMemo(() => {
+    const { title } = simplifyProperties(occurredInPage.properties);
+
+    return title;
+  }, [occurredInPage]);
+
+  return (
+    <TableRow>
+      <TableCell
+        sx={{
+          a: {
+            fontWeight: 600,
+            color: ({ palette }) => palette.blue[70],
+            "&:hover": { color: ({ palette }) => palette.blue[90] },
+          },
+        }}
+      >
+        <Link noLinkStyle href={`/@${triggeredByUser.shortname}`}>
+          {triggeredByUser.preferredName}
+        </Link>{" "}
+        {kind === "new-comment"
+          ? "commented on "
+          : kind === "comment-reply"
+          ? "replied to your comment on "
+          : kind === "page-mention"
+          ? "mentioned you in "
+          : "mentioned you in a comment on "}
+        <Link noLinkStyle href={pageHref ?? ""}>
+          {pageTitle}
+        </Link>{" "}
```

Comment:
Suggest that a user clicking on the link also makes the notification as read

Reply:
Good idea, have addressed in https://github.com/hashintel/hash/pull/3511/commits/4f348327ac60c47cc7859a088309f04d6b45da92